src/Conf.ml

```diff
@@ -1101,9 +1118,11 @@ let xdg_config =
   | None -> None
 
 let build_config ~filename =
-  let files, project_root =
-    collect_files ~dir:(Filename.dirname (to_absolute filename)) []
+  let dir =
+    filename |> to_absolute |> Filename.dirname |> Fpath.v
+    |> Fpath.normalize |> Fpath.to_string
```

Comment:
nit: It might be clearer to use Fpath.segs and express the search up the file system in terms of the list of strings rather than repeatedly performing filename manipulations.